test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -283,9 +283,9 @@ public void CheckToOtlpLogRecordEventId()
         otlpLogRecordAttributes = otlpLogRecord.Attributes.ToString();
 
         // Event
-        Assert.Contains("Id", otlpLogRecordAttributes);
+        Assert.Contains("dotnet.ilogger.event_id", otlpLogRecordAttributes);
         Assert.Contains("10", otlpLogRecordAttributes);
-        Assert.Contains("Name", otlpLogRecordAttributes);
+        Assert.Contains("dotnet.ilogger.event_name", otlpLogRecordAttributes);
```

Comment:
Why not using "event.name"? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/event-api.md#emit-event

Reply:
The intention is to clearly show where this attribute is coming from hence the attribute has `ilogger` in it.

Reply:
This is something @alanwest and I have chatted a lot about and a topic we have discussed on the log SIG meeting a few times. Basically, should things that look like events use the "Event API" as the spec defines it? For example, EventSource also has event name & id should those use the event semantic conventions? So far the thinking has been these are NOT events as the spec defines them, just things that happen to use similar terms, and they should use names tied to the logging framework emitting them (dotnet.ilogger.event_name, dotnet.eventsource.event_name, etc.) NOT the event semantic conventions (event.name). 🤷

Reply:
Bear with me, taking a bit of a tangent here for a sec...

In our prior two SIG meetings, we discussed closing #3491 and leaving our handling of CategoryName in the OTLP exporter. I've more or less conceded to the opinions of others, however for the record I've voiced that I hate the idea of leaving this behavior in the OTLP exporter 😆.

I'm going to make one last ditch effort to reemphasize my original stance on #3491. That is: Abandon the use of CategoryName from all SDK components. In other words, the OTLP exporter will not in anyway export CategoryName. The only way for it to be exported would be for some component upstream to add ILogger's category as a proper attribute.

As this applies to this PR, I think we should abandon the use of EventId from all SDK components.

This has the benefit of not requiring us to settle on some `dotnet.ilogger.*` convention today. Mind you, settling on semantic conventions for logging frameworks should be a very carefully considered effort. I'd far prefer not prioritizing this effort right now in haste.

Reply:
If the goal is to get the OtlpLogExporter stable, I think a reasonable approach would be:
1. by default, only expose features that are covered by the stable version of the spec.
2. for features or any data (e.g. CategoryName, EventId, EventName) that are not covered by the stable spec, either remove them or use feature flags.

Personally I want to encourage feature flags, these data are provided by the user so IMHO they should be observed rather than dropped.

Reply:
To clarify, I think it is fine for `CategoryName` and `EventId` to continue to be set on our `LogRecord`. They just will not be exported.

Reply:
Agree feature flagging seems like a good tactic here.

Reply:
If everyone here is ok with these names, I would like to keep this PR as is and introduce the flag part in separate PR.

@alanwest I am not clear on `feature flagging a behavior of the OTLP exporter that I strongly feel should never belong as a feature of the OTLP exporter is the wrong approach`. As per my understanding exporters can always have their own behavior based on the protocol/backend so not sure how would we push this out of exporter. Can discuss this offline or on the separate PR.

Reply:
Here's an order of operations that I am 👍 on:

1. Land this PR
2. Land the implementation of the feature flag in the OTLP exporter, for now. A mouth full of a suggestion for the name of environment variable gating the feature: `OTEL_DOTNET_EXPERIMENTAL_ADD_ILOGGER_CATEGORYNAME_AND_EVENTID_AS_ATTRIBUTES`
    * Side note: I think 1 and 2 should be done prior to our next beta release.
3. At a later date move all of this functionality out of the OTLP exporter along with its feature flag.

Reply:
> At a later date move all of this functionality out of the OTLP exporter along with its feature flag.

@alanwest When you say move this functionality out of the OTLP exporter, do you mean these attributes should be added in the SDK instead? If yes, I think there some perf concerns that were raised by @CodeBlanch regarding this.

Reply:
> If everyone here is ok with these names, I would like to keep this PR as is and introduce the flag part in separate PR.

I'm not okay, I want to see discussions around the name, e.g. why would we want this vs. that, and the discussion should be backed by user scenarios.

Reply:
> When you say move this functionality out of the OTLP exporter, do you mean these attributes should be added in the SDK instead? If yes, I think there some perf concerns that were raised by @CodeBlanch regarding this.

Yes this is what I mean. Especially since in our SIG meeting discussions it seemed that folks really liked `ilogger` in the attribute names. ILogger specific instrumentation or nomenclature has no place in the OTLP exporter in my opinion.

I had suggested we consider a more general attribute names. Something like `dotnet.logrecord.event_id`. But to @reyang's point whatever we decide we need to carefully consider it and because of that we should not ship this hastily.

I'd have been fine feature flagging the names with `ilogger` in them, but if @reyang you feel that a deeper discussion about names is required even to ship them feature flagged then #4762 just removes the attributes altogether in our stable release. As I voiced earlier, I actually preferred removing them for now anyways.

Reply:
I support removing it from the exporter for stable release.

Reply:
> I'd have been fine feature flagging the names with `ilogger` in them, but if @reyang you feel that a deeper discussion about names is required even to ship them feature flagged then #4762 just removes the attributes altogether in our stable release. As I voiced earlier, I actually preferred removing them for now anyways.

I don't feel we should spend more time on this topic right now, since the high order bit is to ship stable version. Agree with what you said @alanwest, here is the sequence in my mind:
1. For anything that's not covered by the spec or debatable - remove those features.
2. Ship stable version of OTLP Exporter.
3. Gather feedback, and based on the feedback, add experimental features via feature flagging or experimentation, take more feedback.
4. Once we got a good amount of feedback, come back to the discussion and decision making (we don't want to keep discussing something without new inputs/feedback).
5. Once the feature got validated by the users and we also feel comfortable about the direction, ship it as stable (remove the flag).

Reply:
We discussed this in the SIG meeting yesterday. Here is the plan:

1) We will go ahead with this change in favor of resolving the issue faced by user in https://github.com/open-telemetry/opentelemetry-dotnet/issues/4404
2) These attributes will only be added if user enables them via experimental feature flag (to be part of separate PR)

**NOTE**: The names and handling of these attributes are not to be treated as the final design and the discussion around it can continue on https://github.com/open-telemetry/opentelemetry-dotnet/issues/4776.